appinventor/components/src/com/google/appinventor/components/common/YandexLang.java

```diff
@@ -0,0 +1,131 @@
+// -*- mode: java; c-basic-offset: 2; -*-
+// Copyright 2020 MIT, All rights reserved
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package com.google.appinventor.components.common;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Defines a YandexLang type used by the YandexTranslate component. The values
+ * are ISO 639-1 (sometimes 639-2) codes.
+ */
+public enum YandexLang implements OptionList<String> {
```

Comment:
Recommend changing the name of this to `Language` so it isn't specific to Yandex.

Reply:
It actually is specific to Yandex. It only includes the subset of languages that Yandex supports. https://yandex.com/support/translate/supported-langs.html Other components may have different subsets of languages they support, so I thought it might be better to indicate that this was for yandex specifically.

What's your opinion on it though?

Reply:
Those components would still reject an invalid language, so I think it would be fine. We may want to have the languages contain both their ISO 639-1 and ISO-639-2 tags so that components/extensions can take a Language block and get the necessary tag. We could employ filtering similar to what you implemented for assets if the component takes a strict subset of the languages.

Reply:
TextToSpeech as an example.

Reply:
> We may want to have the languages contain both their ISO 639-1 and ISO-639-2 tags so that components/extensions can take a Language block and get the necessary tag. We could employ filtering similar to what you implemented for assets if the component takes a strict subset of the languages.

Hmm given these extra features I think it may be better to implement Languages as their own annotation & helper block, rather than using the dropdowns for them. We could add a property to the annotation specifying which kind of ISO tags the function prefers. We could also add the filtering without having to clutter up the OptionList definition with that.

How would you feel about investigating that approach?

Reply:
We could consider having LanguageISO1 and LanguageISO2, although given that the languages themselves are the "base" object and having different serializations (ISO 639-1 vs ISO 639-2). I would say this isn't critical to finishing the project so let's table it for now and decide the design later once the rest has been merged.